google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -43,45 +43,85 @@ void TracingMessageBatch::SaveMessage(pubsub::Message m) {
 
 namespace {
 
-opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeParentSpan(
-    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
-        message_spans) {
-  using opentelemetry::trace::SpanContext;
-  using AttributesList =
-      std::vector<std::pair<opentelemetry::nostd::string_view,
-                            opentelemetry::common::AttributeValue>>;
-  auto constexpr kMaxOtelLinks = 128;
-  std::vector<std::pair<SpanContext, AttributesList>> links;
-  auto batch_size = message_spans.size();
-  links.reserve(batch_size);
-  // If the batch size is less than the max size, add the links to a single
-  // span.
-  if (batch_size < kMaxOtelLinks) {
-    std::transform(
-        message_spans.begin(), message_spans.end(), std::back_inserter(links),
-        [i = static_cast<std::int64_t>(0)](auto const& span) mutable {
-          return std::make_pair(
-              span->GetContext(),
-              AttributesList{{"messaging.pubsub.message.link", i++}});
-        });
-  }
-  auto batch_sink_parent_span =
+using Spans =
+    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>;
+using Attributes =
+    std::vector<std::pair<opentelemetry::nostd::string_view,
+                          opentelemetry::common::AttributeValue>>;
+using Links =
+    std::vector<std::pair<opentelemetry::trace::SpanContext, Attributes>>;
+
+/// Creates a link for each span in the range @p begin to @p end.
+auto MakeLinks(Spans::const_iterator begin, Spans::const_iterator end) {
+  Links links;
+  std::transform(begin, end, std::back_inserter(links),
+                 [i = static_cast<std::int64_t>(0)](auto const& span) mutable {
+                   return std::make_pair(
```

Comment:
`#include <iterator>` for `std::back_inserter` and `#include <utility>` for `std::make_pair`.